test/dynamo/test_optimizers.py

```diff
@@ -25,15 +25,26 @@ def make_test(optim_cls, closure=None, **kwargs):
 
     def test_fn(self):
         nonlocal opt
+
+        # run the patcher so that step has the expected structure
+        torch._dynamo.eval_frame.TorchPatcher.patch()
+
+        # unwrap step to avoid a deliberate graph break due to
+        # a limitation of functionalization/no_grad detection
+        # see the comment in optimizer.py
+        step_fn = opt.step.__wrapped__
```

Comment:
I think it is worth pointing that the you're unwrapping step from within `_use_grad_for_differentiable`, which is used by differentiable optimizers today and not supported in PT2.